docs/content/docs/reference/pubsub.md

```diff
@@ -42,5 +45,108 @@ Events can be published to a topic like so:
 Invoices.Publish(ctx, Invoice{...})
 ```
 
+<!-- kotlin -->
+
+First, declare a new topic :
+
+```kotlin
+@Export
+@TopicDefinition("invoices")
+internal interface InvoiceTopic : Topic<Invoice>
+```
+
+Events can be published to a topic by injecting it into an `@Verb` method:
+
+```kotlin
+@Verb
+fun publishInvoice(request: InvoiceRequest, topic: InvoiceTopic) {
+    topic.publish(Invoice(request.getInvoiceNo()))
+}
+```
+
+There are two ways to subscribe to a topic. The first is to declare a method with the `@Subscription` annotation, this is generally used when
+subscribing to a topic inside the same module:
+
+```kotlin
+@Subscription(topic = "invoices", name = "invoicesSubscription")
+fun consumeInvoice(event: Invoice) {
+    // ...
+}
+```
+
+This is ok, but it requires the use of string constants for the topic name, which can be error-prone. If you are subscribing to a topic from
+another module, FTL will generate a type-safe subscription meta annotation you can use to subscribe to the topic:
```

Comment:
I _really_ don't love this as an end state, is there a way we can have type safety for topics in the same module too?

Reply:
I think you could just directly reference the topic in this case, I will put up a PR next week.